### PR TITLE
Ambient light improvements

### PR DESCRIPTION
**Pull request by appgurueu**
* Remove redundant luminance. This makes it both more straightforward for modders and the engine.
* Ensure that unused ambient light alpha is always 255. (It might be cleaner to use a `std::array<u8, 3>` instead. It would be cleanest to have a RGB color class in Irrlicht, but ain't nobody got time for that.)
* I've also "fixed" the set lighting packet reading to use early returns to avoid some theoretical shenanigans (and checked for the proper amount of bytes).

### CODE REVIEW

**Comment by Andrey2470T** [comment]
Thanks. I will merge your changes now. However, I'm not sure whether the error should be thrown out if the passed alpha is not 255. Actually it is just not used anywhere, so I think the passing any alpha won't affect anything.

**Comment by appgurueu** [comment]
> Thanks. I will merge your changes now. However, I'm not sure whether the error should be thrown out if the passed alpha is not 255. Actually it is just not used anywhere, so I think the passing any alpha won't affect anything.

I think it's good to be strict. If someone passes an alpha different from 255, they are probably expecting something to happen with that. A stricter API can always be relaxed later on - who knows, we might have a use for the alpha channel in the future. The opposite - making a lax API stricter - is much harder; you have to figure out how the lax API is used and why, and then deprecate that, so that you can eventually change the functionality.

**TL;DR**: I don't think we should design our APIs to be any more permissive than necessary, unless there are tangible benefits, and I don't see a good benefit to passing a non-255-alpha.
